drivers/homeless_summary_report/app/controllers/homeless_summary_report/warehouse_reports/reports_controller.rb

```diff
@@ -0,0 +1,169 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HomelessSummaryReport::WarehouseReports
+  class ReportsController < ApplicationController
+    include WarehouseReportAuthorization
+    include AjaxModalRails::Controller
+    include ArelHelper
+    include BaseFilters
+
+    before_action :require_can_access_some_version_of_clients!, only: [:details]
+    before_action :set_report, only: [:show, :destroy, :details]
+    before_action :set_variants, only: [:show, :details]
+    before_action :set_m1_fields, only: [:show, :details]
+    before_action :set_m2_fields, only: [:show]
+    before_action :set_m7_fields, only: [:show, :details]
+
+    def index
+      @reports = report_scope.ordered.
+        page(params[:page]).per(25)
+      @report = report_class.new(user_id: current_user.id)
+      # Make sure the form will work
+      filters
+    end
+
+    def show
+      @measures = {
+        'Measure 1': @m1_fields.keys,
+        'Measure 2': @m2_fields,
+        'Measure 7': @m7_fields,
+      }
+    end
+
+    def create
+      @report = report_class.new(
+        user_id: current_user.id,
+      )
+      @report.filter = @filter
+      @report.save
+      # TODO: Now we need to figure out how to define the job. I think it's by modifying the report class, probably ultimately the run_and_save! method. Yes.
+      ::WarehouseReports::GenericReportJob.perform_later(
+        user_id: current_user.id,
+        report_class: @report.class.name,
+        report_id: @report.id,
+      )
+      # Make sure the form will work
+      filters
+      respond_with(@report, location: homeless_summary_report_warehouse_reports_reports_path)
+    end
+
+    def destroy
+      @report.destroy
+      respond_with(@report, location: homeless_summary_report_warehouse_reports_reports_path)
+    end
+
+    def details
+      @variant = details_params['variant'] || 'all_persons'
+      @cell = "spm_#{details_params['cell']}" || 'spm_m1a_es_sh_days'
```

Comment:
The || here will never return nil since you are concatenating a string, so `'spm_m1a_es_sh_days'` will never be used.  If `details_params['cell']` is blank you'll get `spm_`.